data/pigui/game.lua

```diff
@@ -0,0 +1,512 @@
+	local offset = 30
+	local width = 10
+	local height_hrs = 0.1
+	local hrs = roll_degrees / 360 * 12 + 3
+	-- left hook
+	ui.addLine(ui.pointOnClock(center, reticuleCircleRadius - offset, hrs),
+						 ui.pointOnClock(center, reticuleCircleRadius - offset - width, hrs),
+						 ui.theme.colors.navigationalElements, 1)
+	ui.addLine(ui.pointOnClock(center, reticuleCircleRadius - offset, hrs),
+						 ui.pointOnClock(center, reticuleCircleRadius - offset, hrs + height_hrs),
+						 ui.theme.colors.navigationalElements, 1)
+	ui.addLine(ui.pointOnClock(center, reticuleCircleRadius - offset, -3),
+						 ui.pointOnClock(center, reticuleCircleRadius - offset + width/2, -3),
+						 ui.theme.colors.navigationalElements, 1)
+	-- right hook
+	ui.addLine(ui.pointOnClock(center, reticuleCircleRadius - offset, hrs + 6),
+						 ui.pointOnClock(center, reticuleCircleRadius - offset - width, hrs + 6),
+						 ui.theme.colors.navigationalElements, 1)
+	ui.addLine(ui.pointOnClock(center, reticuleCircleRadius - offset, hrs + 6),
+						 ui.pointOnClock(center, reticuleCircleRadius - offset, hrs + 6 - height_hrs),
+						 ui.theme.colors.navigationalElements, 1)
+	ui.addLine(ui.pointOnClock(center, reticuleCircleRadius - offset, 3),
+						 ui.pointOnClock(center, reticuleCircleRadius - offset + width/2, 3),
+						 ui.theme.colors.navigationalElements, 1)
+
+
+
+end
+
+local function displayReticuleCompass(center, heading)
+	local relevant = {}
+	local directions = { [0] = lc.COMPASS_N, [45] = lc.COMPASS_NE, [90] = lc.COMPASS_E, [135] = lc.COMPASS_SE, [180] = lc.COMPASS_S, [225] = lc.COMPASS_SW, [270] = lc.COMPASS_W, [315] = lc.COMPASS_NW }
+	local function cl(x)
+		if x < 0 then
+			return cl(x + 360)
+		elseif x >= 360 then
+			return cl(x - 360)
+		else
+			return x
+		end
+	end
+	local left = math.floor(heading - 45)
+	local right = left + 90
+	local d = left
+
+	local function stroke(d, p, n, height, thickness)
+		if d % n == 0 then
+			local a = ui.pointOnClock(center, reticuleCircleRadius, 2.8 * p - 1.4)
+			local b = ui.pointOnClock(center, reticuleCircleRadius + height, 2.8 * p - 1.4)
+			ui.addLine(a, b, ui.theme.colors.reticuleCircle, thickness)
+		end
+	end
+
+	ui.addLine(ui.pointOnClock(center, reticuleCircleRadius, 0),
+						 ui.pointOnClock(center, reticuleCircleRadius - 3, 0),
+						 ui.theme.colors.reticuleCircle, 1)
+
+	while true do
+		if d > right then
+			break
+		end
+		local p = (d - left) / 90
+		stroke(d, p, 15, 3, 1)
+		stroke(d, p, 45, 4, 1)
+		stroke(d, p, 90, 4, 2)
+		for k,v in pairs(directions) do
+			if cl(k) == cl(d) then
+				local a = ui.pointOnClock(center, reticuleCircleRadius + 8, 3 * p - 1.5)
+				ui.addStyledText(a, v, ui.theme.colors.navigationalElements, ui.fonts.pionillium.tiny, ui.anchor.center, ui.anchor.bottom, "")
+			end
+		end
+		d = d + 1
+	end
+end
+
+local function displayReticuleDeltaV(center)
+	-- ratio is 1.0 for full, 0.0 for empty
+	local function gauge(ratio, center, radius, color, thickness)
```

Comment:
This (and the latter function of the same name) are specialised to left/right, they are not exactly the same. They should be refactored, I just haven't found a very good way to do that yet :-/ I'll probably try to extract much of game.lua into a generic "GUI" toolkit later.